npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -253,13 +229,26 @@ func (pMgr *PolicyManager) runIPTablesCommand(operationFlag string, args ...stri
 	return 0, nil
 }
 
-func (pMgr *PolicyManager) creatorForInitChains() *ioutil.FileCreator {
-	creator := pMgr.newCreatorWithChains(iptablesAzureChains)
+// Writes the restore file for bootup, and marks the following as stale: deprecated chains and old v2 policy chains.
+// This is a separate function to help with UTs.
+func (pMgr *PolicyManager) creatorForBootup(currentChains map[string]struct{}) *ioutil.FileCreator {
+	pMgr.staleChains.empty()
+	chainsToCreate := make([]string, 0, len(iptablesAzureChains))
+	for _, chain := range iptablesAzureChains {
+		_, exists := currentChains[chain]
+		if !exists {
+			chainsToCreate = append(chainsToCreate, chain)
+		}
+	}
 
-	// add AZURE-NPM chain rules
-	creator.AddLine("", nil, util.IptablesAppendFlag, util.IptablesAzureChain, util.IptablesJumpFlag, util.IptablesAzureIngressChain)
-	creator.AddLine("", nil, util.IptablesAppendFlag, util.IptablesAzureChain, util.IptablesJumpFlag, util.IptablesAzureEgressChain)
-	creator.AddLine("", nil, util.IptablesAppendFlag, util.IptablesAzureChain, util.IptablesJumpFlag, util.IptablesAzureAcceptChain)
+	// Step 2.1 in bootup() comment: cleanup old NPM chains, and configure base chains and their rules
+	// To leave NPM deactivated, don't specify any rules for AZURE-NPM chain.
```

Comment:
I see we do not duplicate rules here, only activate it when the first policy is applied.